src/NewProject/add_constraints_form.cpp

```diff
@@ -34,3 +37,16 @@ bool addConstraintsForm::IsCopySource() {
   return ui->m_ckkBoxCopy->checkState() == Qt::CheckState::Checked ? true
                                                                    : false;
 }
+
```

Comment:
This code looks problematic. You're assign on top of the parameter being passed in. Either check what is already being reported by the signal by checking define_Order_Checked or if you need to most current value from the widget then just check against ui->select_defineOrder->isChecked(). Same goes for random_checked, the assignment is unneccesary and might actually cause some unintended effects.

Reply:
On second look, on_select_defineOrder_clicked and on_select_random_clicked seem to be unnecessary. The settings radiobuttons widget is already mutually exclusive (when you click 1 radio button, the currently selected option gets de-selected) so you don't need to toggle the two values back and forth. If that's all your logic was intending here then i'd just remove both these and see if your code still works as expected.

Reply:
Yeah You are right. I think should work.